internal/prometheus/prometheus_test.go

```diff
@@ -4,8 +4,8 @@
 
 // NOTE(sr): Different go runtime metrics on 1.19.
 // This can be removed when we drop support for go 1.19.
-//go:build go1.19
-// +build go1.19
+//go:build go1.19 && !go1.20
+// +build go1.19,!go1.20
 
```

Comment:
This makes the test pass but we probably need to pull in https://github.com/prometheus/client_golang/tree/v1.14.0/prometheus/collectors.